### ISSUE TITLE
Unable to Achieve Custom Width in Custom Component Class

### ISSUE THREAD

**Issue by neilchaudhuri** [tailwindcss/discuss#372]
I would like to add a responsive custom width to a custom component class. 

My (simplified) configuration looks like this:

```
spacing: {
      px: '1px',
      '0': '0',
      '1': '0.25rem',
      '2': '0.5rem',
      ...
      '16': '4rem',
      '18': '4.5rem' // This is custom
      ...
    }
```

My CSS looks like this:

```
@tailwind base;

@tailwind components;

.btn {
    @apply h-12 lg:w-18 m-0 p-0
}

@tailwind utilities;

```

I get this error:

```
 `@apply` cannot be used with `.lg\:w-18` because `.lg\:w-18` either cannot be found, or its actual definition includes a pseudo-selector like :hover, :active, etc. If you're sure that `.lg\:w-18` exists, make sure that any `@import` statements are being properly processed *before* Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.
```

Obviously I am missing something in how to configure things. Any help is appreciated.

**Comment by adamwathan** [comment]
You've got it configured correctly but you can't use `@apply` with responsive prefixes yet.

Explained in the docs here:

https://tailwindcss.com/docs/functions-and-directives#apply

TL;DR write it like this:

@tailwind base;

@tailwind components;

.btn {
    @apply h-12 m-0 p-0
}

@screen lg {
  .btn {
    @apply w-18
  }
}

@tailwind utilities;

**Comment by neilchaudhuri** [comment]
Ah missed that in the documentation. Thanks for the prompt response.
